src/parser/Parser.cpp

```diff
@@ -3,98 +3,105 @@
 Parser::Parser(const Lexer lexer)
     : m_lexer(lexer)
 {
+    m_currenttoken = m_lexer.getToken();
+}
 
+Token Parser::eattoken(TokenType t) {
+    if (m_currenttoken.type != t) {
+        puts("Unexpected token type.");
+        exit(1);
+    }
+    m_currenttoken = m_lexer.getToken();
 }
 
 std::unique_ptr<AST::Stmt> Parser::parse() {
     return parseStmt();
 }
 
 std::unique_ptr<AST::Stmt> Parser::parseStmt(){
-    Token currTok = m_lexer.getToken();
-    if (currTok.type == TokenType::EOL || currTok.type == TokenType::END) {
```

Comment:
We'll still need this, no? Otherwise we'll crash as soon as we reach the end of the file.